commands/serve/web/hooks/useConnection.js

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect, useMemo } from 'react';
+import { manageConnections } from '../utils';
+import { connect } from '../connect';
+import storageFn from '../storage';
+
+// TODO:
+// when we would get session property on enigma instance?
+// this seems to be the only case when we return enigma instace
+export const handleSessionNotification = ({ result, setError, setGlobal }) => {
+  if (result.session) {
```

Comment:
Do we need this check? Is it necessary to have this one in here or is it some kind of legacy code that has been forgotten to be removed?

Reply:
if this is only called on a successful connection then it should not be needed no. The answer probably lies in what happens if the session is undefined, what does it do then?

Reply:
the only case that is possible to being invoked (not 100% sure if that is being called) is by connection to local engine in our machine, because that is the only case when we get the react instance of enigma and it has the `session.on()` function in it. 

But still i'm pretty unsure that will we (or when we) get this event of `notification:OnAuthenticationInformation` in enigma flow + when we would fall in the error case of that event which has `mustAuthenticate` flag.